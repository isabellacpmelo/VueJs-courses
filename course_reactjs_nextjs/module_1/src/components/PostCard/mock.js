export const postCardProps = {
    id: '1',
    title: 'title mock',
    body: 'body mock',
    cover: 'img/img.png'
}