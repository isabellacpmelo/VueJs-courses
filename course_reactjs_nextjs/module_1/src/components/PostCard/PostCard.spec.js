/* eslint-disable testing-library/no-node-access */
import { render, screen} from '@testing-library/react'
import { PostCard } from '.'
import { postCardProps } from './mock'

const props = postCardProps

describe('<PostCard />', () => {
    test('should render PostCard correctly', () => {
        render(<PostCard {...props} />)

        expect(screen.getByAltText(/title mock/i))
            .toHaveAttribute('src', 'img/img.png')

        expect(screen.getByRole('heading', { name: /title mock/i }))
            .toBeInTheDocument()

        expect(screen.getByText('body mock'))
            .toBeInTheDocument()
    })


    test('should match snapshot', () => {
        const { container } = render(<PostCard {...props} />)

        expect(container.firstChild).toMatchSnapshot()
    })
})