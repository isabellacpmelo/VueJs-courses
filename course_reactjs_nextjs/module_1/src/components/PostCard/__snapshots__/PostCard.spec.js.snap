// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title mock"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      title mock
    </h2>
    <p>
      body mock
    </p>
  </div>
</div>
`;
